elasticsearch/src/main/scala/com/snowplowanalytics/stream/loader/clients/ElasticsearchBulkSender.scala

```diff
@@ -97,6 +99,21 @@ class ElasticsearchBulkSender(
             new ElasticsearchObject(documentIndex, documentType, compact(render(r.json)))
         }
     }
+
+    // blindly creating the index, the error is ignored if the index would exists
+    successfulRecords.map(_.getIndex).distinct.map(index => {
+      val createIndexRequest = createIndex(index)
+      shardsCount match {
+        case Some(value) => createIndexRequest.shards(value)
+        case None =>
+      }
+      replicasCount match {
+        case Some(value) => createIndexRequest.replicas(value)
+        case None =>
+      }
+      client.execute(createIndexRequest).await
+    })
```

Comment:
mmh we don't want to create an index every time we send records, do we?

Reply:
I thought about it for a while, too. I have three points on this:

- We could first check whether the index exists and then create them. 
- We could keep track of the ones that we have been created but if we have multiple instances of this application running on the same queue to have more throughput, one will create the index and the others fails when creating. So parhaps this approach might be mixed with the one above.
- I needed to submit this PR to open my way for another one. I was thinking about storing static mapping for the index, too.

The other ways is to use Index Template and close this PR. Probably this might be a better approach to keep this application with less settings to maintain: https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-templates.html#indices-templates

Please feel free to close this PR if you think Index Template is a better approach. I want to work on the sharding based on the time (Sharded Indices by Date)